### DISCUSSION TITLE
Does codecarbon work for Windows computers using AMD CPU?

### DISCUSSION THREAD

**Discussion by rosekelly6400** [mlco2/codecarbon#639] [Q&A]
Hello, I'm trying to understand the CPU emissions tracking methods codecarbon employs and this is my rough understanding:

Mac:
   - Apple Silicon Chips: Powermetrics
   - Intel CPU: Powermetrics OR Intel Power Gadget
   - AMD CPU: N/A

Windows:
   - Intel CPU: Intel Power Gadget
   - AMD CPU: ???

Linux:
   - Intel CPU: RAPL
   - AMD CPU: RAPL
     
So it seems like computers using AMD CPU and running Windows OS is missing here. Is there currently a way to track emissions for Windows with AMD CPU?

**Comment by benoit-cty** [comment]
Hello, sorry for the late answer. Yes I think we lack the ability to measure AMD GPU on Windows.

If you know a way to do it please update this issue https://github.com/mlco2/codecarbon/issues/457
